--- a/include/uapi/linux/ethtool.h
+++ b/include/uapi/linux/ethtool.h
@@ -14,7 +14,6 @@
 #define _UAPI_LINUX_ETHTOOL_H
 
 #include <linux/types.h>
-#include <linux/if_ether.h>
 
 /* All structures exposed to userland should be defined such that they
  * have the same layout for 32-bit and 64-bit userland.
--- a/include/uapi/linux/if_bridge.h
+++ b/include/uapi/linux/if_bridge.h
@@ -14,7 +14,6 @@
 #define _UAPI_LINUX_IF_BRIDGE_H
 
 #include <linux/types.h>
-#include <linux/if_ether.h>
 
 #define SYSFS_BRIDGE_ATTR	"bridge"
 #define SYSFS_BRIDGE_FDB	"brforward"
